samples/manifest-request.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+async function main(
+  projectId = 'my-project',
+  sourceAgentPoolName = '',
+  rootDirectory = '',
+  gcsSinkBucket = '',
+  manifestLocation = ''
+) {
+  // [START storagetransfer_manifest_request]
+
+  // Imports the Google Cloud client library
+  const {
+    StorageTransferServiceClient,
+  } = require('@google-cloud/storage-transfer');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // Your project id
+  // const projectId = 'my-project'
+
+  // The agent pool associated with the POSIX data source. Defaults to the default agent
+  // const sourceAgentPoolName = 'projects/my-project/agentPools/transfer_service_default'
+
+  // The root directory path on the source filesystem
+  // const rootDirectory = '/directory/to/transfer/source',
+
+  // The ID of the GCS bucket to transfer data to
+  // const gcsSinkBucket = 'my-sink-bucket'
+
+  // Transfer manifest location. Must be a
+  // const manifestLocation = 'gs://my-bucket/sample_manifest.csv'
+
+  // Creates a client
+  const client = new StorageTransferServiceClient();
+
+  /**
+   * Creates a request to transfer from the local file system to the sink bucket
+   */
+  async function transferDirectory() {
```

Comment:
Is "transferDirectory" the right name for this function? This only transfers the files specified in the manifest right? which may not be the entire directory if i understand it correctly, and may in fact include files from multiple directories within the root directory